### PR TITLE
feat: add getGasPerPubdataByte endpoint

### PR DESCRIPTION
**Pull request by dimazhornyk**
## What ❔

Add a `getGasPerPubdataByte` endpoint in `zks` namespace, that returns the amount of L2 gas required to pay for 1 byte of pubdata.

## Why ❔

One of our partners requires this endpoint, they want to have a better insight into the pricing model.

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [ ] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

File: core/lib/zksync_core/src/api_server/web3/namespaces/zks.rs [hunk 1]
```diff
@@ -571,4 +571,18 @@ impl ZksNamespace {
             .base_token_address
             .ok_or(Web3Error::NotImplemented)
     }
+
+    #[tracing::instrument(skip(self))]
+    pub async fn get_gas_per_pubdata_byte_impl(&self) -> U256 {
```
Comment (slowli) [inline core/lib/zksync_core/src/api_server/web3/namespaces/zks.rs:576]: Dumb question: Would it be unreasonable to have a method returning `PubdataIndependentBatchFeeModelInput`, which contains data necessary to compute the required value, and also would obsolete the adjacent `zks_getL1GasPrice` method? IMO, it's better to have fewer methods with greater utility than a multitude of overspecialized methods.

Comment (dimazhornyk) [reply core/lib/zksync_core/src/api_server/web3/namespaces/zks.rs]: Great point, I did exactly that. Please, have a look and check whether it's what you were thinking

**Review by slowli — APPROVED** [review]
Looks good! Keep in mind that [a recent PR](https://github.com/matter-labs/zksync-era/pull/1793) has made `batch_fee_input_provider` fallible, but it should be easy to update this PR.

**Comment by dimazhornyk** [comment]
@slowli I resolved the conflicts, could you please approve it once again?

**Pull request merged**